src/wast-lexer.cc

```diff
@@ -32,27 +32,77 @@
 
 #define INITIAL_LEXER_BUFFER_SIZE (64 * 1024)
 
-#define YY_USER_ACTION                  \
-  {                                     \
-    loc->filename = filename_;          \
-    loc->line = line_;                  \
-    loc->first_column = COLUMN(token_); \
-    loc->last_column = COLUMN(cursor_); \
-  }
+#define PUSH_LOOKAHEAD_TOKEN              \
```

Comment:
As long as you're cleaning this stuff up... :-)

I wonder if some of these macros can be member functions of WastLexer instead? Or maybe even primarily functions with a small macro wrapper for name conversion. Wdyt?